src/main/resources/db/migration/V1_132__print_files_to_clob_update.sql

```diff
@@ -0,0 +1,216 @@
+
+	FOR forms IN
+		SELECT	fr.form_type,
+				fr.max_rec_len::integer
+		FROM juror_mod.t_form_attr fr
+
+		loop
+			CALL juror_mod.write_printfiles(forms.form_type, forms.max_rec_len, v_ext_date, p_document_limit);
+		END LOOP;
+
+END;
+$procedure$
+;
+
+DROP PROCEDURE juror_mod.write_printfiles(in varchar, in int4, in date, in int4, inout int4);
+
+CREATE OR REPLACE PROCEDURE juror_mod.write_printfiles(IN p_form_type character varying, IN p_max_rec_len integer, IN p_ext_date date, IN p_document_limit integer)
+ LANGUAGE plpgsql
+AS $procedure$
+DECLARE
+	form_details RECORD; -- current record from bulk print data table.
+	v_filename VARCHAR(50); -- current filename
+	v_header VARCHAR(50); -- header for print file
+	v_row_id INTEGER; -- row id of current record from bulk print data
+	v_data text:=''; -- data to be written to file
+	v_detail_rec text; -- row of data from bulk print data
+	v_revision int8; -- sequence number for file name
+	v_page_size integer; -- total pages for form type
+	v_total_documents_size integer; -- total amount of remaining records to write
+	v_documents_to_write integer; -- total amount of documents to write to current file
+
+begin
+	select count(bpd.id)
+	from juror_mod.bulk_print_data bpd
+	WHERE bpd.form_type = p_form_type
+	AND COALESCE(bpd.extracted_flag,'N') = 'N'
+	AND DATE(bpd.creation_date) <= p_ext_date
+	into v_total_documents_size;
+
+	raise notice 'total_docs_size %', v_total_documents_size;
+
+	-- Identify the next sequence number
+	SELECT  NEXTVAL('juror_mod.print_file_count') INTO v_revision;
+	v_filename := 'JURY'||LPAD(v_revision::VARCHAR(22),4,'0')||'01.0001';
+
+	call juror_mod.get_form_type_page_length(p_form_type, v_page_size);
+
+
+	-- Calulate documents to write
+	if v_total_documents_size - p_document_limit < 0 then
+		v_documents_to_write := v_total_documents_size;
+	else
+		v_total_documents_size := v_total_documents_size - p_document_limit;
+		v_documents_to_write := p_document_limit;
+	end if;
+
+	FOR form_details IN
+		SELECT 	ROW_NUMBER() OVER () as counter,
+				bpd.id,
+				REPLACE(REPLACE(bpd.detail_rec,CHR(10),' '),CHR(13),' ') detail_rec
+		FROM juror_mod.bulk_print_data bpd
+		WHERE bpd.form_type = p_form_type
+		AND COALESCE(bpd.extracted_flag,'N') = 'N'
+		AND DATE(bpd.creation_date) <= p_ext_date
+	loop
+
+		-- setting header for file first time only
+		IF v_header IS null or v_header = '' THEN
+			SELECT RPAD('   ' ||RPAD(p_form_type,16)||LPAD(v_documents_to_write::VARCHAR(20),6,'0')||LPAD(v_page_size::VARCHAR(20),6,'0')||'50'||LPAD(p_max_rec_len::VARCHAR(20),8,'0'),256,' ')
+			INTO v_header;
+			v_data := '' || v_header || chr(10);
+		END if;
+
+		IF p_count >= p_document_limit then
+			-- Create the record
+			INSERT INTO juror_mod.content_store(request_id,document_id,file_type,data)
+			VALUES (NEXTVAL('juror_mod.content_store_seq'),v_filename,'PRINT',v_data);
+			COMMIT;
+
+			-- setting new file name as last file has filled up to the limit
```

Comment:
Are we sure that we will need another file at this point. What if there are no more records to write for this file type after having written the file above i.e. number of records that needed to be extracted exactly matched the file limit.